metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_interfaces_class_diagram.puml

```diff
@@ -3,41 +3,47 @@
     package integration.interface {
         package base {
             package party {
-                interface IntegrationParty<T extends Config, K extends IntegrationClient, V extends ConfigValidator> {
-                    + clientFactory : IntegrationClientFactory<T, K>
-                    + configParser : ConfigParser<T>
-                    + configValidatorFactory : ConfigValidatorFactory<V>
+                interface IntegrationParty<T extends Config, K extends IntegrationClient> {
+                    + clientFactory: IntegrationClientFactory<T, K>
+                    + configParser: ConfigParser<T>
+                    + configValidatorFactory: ConfigValidatorFactory<T>
                 }
             }
 
             package config {
                 package model {
                     interface Config {}
 
-                    package builder {
-                        abstract class ValidationResultBuilder<T extends ValidationResult> {
-                            + authValidationResult : FieldValidationResult
-                            + interruptReason : String
+                    abstract class ConfigField {
+                        + value: String
+
+                        + String toString()
+                    }
 
-                            + T build()
-                            + void setAuthValidationResult(FieldValidationResult authValidationResult)
-                            + void setInterruptReason(String reason)
+                    package builder {
+                        class ValidationResultBuilder {
+                            - _results: Map<ConfigField, FieldValidationResult>
+                            
+                            + ValidationResultBuilder forFields(List<ConfigField> fields)
+
+                            + ValidationResult build()
+                            + void setResult(ConfigField field, FieldValidationResult result)
+                            + void setEmptyResults(FieldValidationResult result)
                         }
                     }
 
                     interface ValidationResult {
-                        + authValidationResult : FieldValidationResult
+                        + results: UnmodifiableMapView<ConfigField, FieldValidationResult>
 
                         + String toString()
                     }
                 }
 
                 package validator {
-                    abstract class ConfigValidator<T extends Config, V extends ValidationDelegate, R extends ValidationResult> {
-                        + validationDelegate : V
-                        + validationResultBuilder : ValidationResultBuilder<R>
+                    interface ConfigValidator<T extends Config> {
```

Comment:
Would it be great to inject the `ValidationResultBuilder` into this class to make it a bit more testable?